rspace/src/main/scala/coop/rchain/rspace/HotStore.scala

```diff
@@ -190,13 +190,13 @@ private class InMemHotStore[F[_]: Sync, C, P, A, K](
       continuations <- getContinuations(join)
       _ <- if (continuations.isEmpty) S.flatModify { cache =>
             val index = cache.joins(channel).indexOf(join)
-            removeIndex(cache.joins(channel), index) >>= { updated =>
+            if (index != -1) removeIndex(cache.joins(channel), index) >>= { updated =>
               Sync[F]
                 .delay {
                   cache.joins.put(channel, updated)
                 }
                 .map(_ => cache)
-            }
+            } else cache.pure[F]
```

Comment:
What does index represent?

Reply:
I guess why does it matter that the index is -1 for removal?

Reply:
It's the index of the element in the sequence. If it's `-1` then the element is not present (as per contract of `indexOf`) and `removeIndex` will raise an error.